### PR TITLE
Feature/move plugins

### PR DESCRIPTION
**Pull request by vutran**
Moving plugins-related functions into a separate module so it is more maintainable.

The new `plugins` module has a new method `enable()`, and `disable()` which encapsulates adding/removing the plugin from `~/.dext/config.json` to enable resolving vutran/dext#64. This method is accessible via:

- `api.plugins.enable(plugin)`
- `api.plugins.disable(plugin)`

In addition, `createSymLink()`, and `removeSymLink()` will now enable and disable the plugin. This resolves vutran/dext-cli#21

### CODE REVIEW

File: src/errors/index.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 exports.ERR_MODULE_DOWNLOAD_ERROR = 'ERR_MODULE_DOWNLOAD_ERROR';
 exports.ERR_MODULE_INSTALLED = 'ERR_MODULE_INSTALLED';
+exports.ERR_MODULE_ENABLED = 'ERR_MODULE_ENABLED';
 exports.ERR_MODULE_NOT_FOUND = 'ERR_MODULE_NOT_FOUND';
+exports.ERR_MODULE_NOT_ENABLED = 'ERR_MODULE_NOT_ENABLED';
```
Comment (adnasa) [inline src/errors/index.js]: Why not `ERR_MODULE_DISABLED`?

Comment (adnasa) [inline src/errors/index.js]: Why not `ERR_MODULE_DISABLED`?

File: src/api/plugins.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+const getAll = () => config.get('plugins') || [];
+
+/**
+ * Checks if the plugin is already enabled
+ *
+ * @param {String} plugin - The plugin name
+ * @return {Boolean}
+ */
+const isEnabled = plugin => getAll().indexOf(plugin) > -1;
```
Comment (adnasa) [inline src/api/plugins.js]: I think this can be simplified

```js 
Boolean(~getAll().indexOf(plugin))
```

Comment (vutran) [inline src/api/plugins.js]: `Boolean()` seems unnecessary but NOT op LGTM. Will add this soon.

**Pull request merged**